cpp/src/dictionary/remove_keys.cu

```diff
@@ -57,13 +58,18 @@ std::unique_ptr<column> remove_keys_fn(
   // create keys positions column to identify original key positions after removing they keys
   auto const keys_view = dictionary_column.keys();
   auto execpol         = rmm::exec_policy(stream);
-  rmm::device_vector<int32_t> keys_positions(keys_view.size());  // needed for remapping indices
+  rmm::device_uvector<uint32_t> keys_positions(keys_view.size(), stream);  // for remapping indices
   thrust::sequence(execpol->on(stream), keys_positions.begin(), keys_positions.end());
   column_view keys_positions_view(
-    data_type{type_id::INT32}, keys_view.size(), keys_positions.data().get());
+    data_type{type_id::UINT32}, keys_view.size(), keys_positions.data());
 
   // copy the non-removed keys ( keys_to_keep_fn(idx)==true )
-  rmm::device_vector<int32_t> map_indices(keys_view.size(), -1);  // init -1 to identify new nulls
+  rmm::device_uvector<uint32_t> map_indices(keys_view.size(), stream);
+  thrust::fill(execpol->on(stream),
+               map_indices.begin(),
+               map_indices.end(),
+               static_cast<uint32_t>(std::numeric_limits<size_type>::max()));
```

Comment:
shouldn't this be 
```suggestion
               static_cast<uint32_t>(std::numeric_limits<uint32_t>::max()));
```

Reply:
No. The upper limit is actually the number of possible elements which is max of `size_type`.

Reply:
This seems like something we should give a name and define in `types.hpp`. Something like:

```
constexpr size_type MAX_COLUMN_SIZE{std::numeric_limits<size_type>::max()};
```

Reply:
Adding `constexpr size_type max_column_size{std::numeric_limits<size_type>::max()};` to `types.hpp` requires also adding `#include <limits>` which causes `gtests/BINARY_TEST` to fail:
https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fcudf%2Fprb%2Fcudf-gpu-build/detail/cudf-gpu-build/32004/pipeline
So I'll just move this definition to file in which I use it for now.